src/platform/nrfconnect/BUILD.gn

```diff
@@ -49,8 +49,19 @@ static_library("nrfconnect") {
     "SystemPlatformConfig.h",
   ]
 
+  public = [ "${chip_root}/src/credentials/DeviceAttestationCredsProvider.h" ]
+
   public_deps = [ "${chip_root}/src/platform:platform_base" ]
 
+  if (chip_enable_factory_data && !chip_enable_factory_data_custom_backend) {
```

Comment:
`chip_enable_factory_data` would be enough imho. It doesn't control whether one can use factory data at all, but only if the default implementation should be enabled.